### PR TITLE
Game Load: Show list of enabled modification.

### PR DESCRIPTION
**Pull request by jostephd**
screenshot_2018-08-24_18-33-20

This PR adds the "Modifications:" header and bulleted list. It shows the mod name if the mod is installed, else mod id, as done for campaigns.

Similar to #3478.

### CODE REVIEW

**Comment by jostephd** [comment]
The dialog would have a vertical scrollbar next to the file list scrollbar. Not pretty but it works.

screenshot_2018-08-24_21-56-13
screenshot_2018-08-24_21-56-49

**Comment by jostephd** [comment]
There's a bug: the vertical space allocated to the mods list is determined by the first savegame. If the second savegame has more mods than the first savegames, the excess ones won't be shown.

**Comment by gfgtdf** [comment]
> The dialog would have a vertical scrollbar next to the file list scrollbar. Not pretty but it works.

I think It'd be better if there would appear a scrollbar in the left-side-panel in that case, and of course also only for those savegames that actuall have too many mods enabled.

**Comment by jostephd** [comment]
Done - screenshot_2018-08-25_16-04-50

**Comment by gfgtdf** [comment]
> Done -

i assume this also fixed the bug you meantione eariler ?

**Comment by jostephd** [comment]
Yes, that bug was fixed by 57afe4d. Sorry for being unclear.

**Comment by gfgtdf** [comment]
does that mean the dialogs size can now change when selecting another game? Usually `window.invalidate_layout` shodul be avoided becasue it causes things like that.

**Comment by jostephd** [comment]
The dialog's size does not change when selecting another game. The only thing that changes is the location of the leader list, but it remains visible at all times. If the `invalidate_layout` were removed, a vertical scrollbar would be used even when a re-layout with no vertical scrollbar was possible (you can see this whenever selecting a savegame that has more mods than the newest savegame).

File: src/gui/dialogs/game_load.cpp [hunk 1]
```diff
@@ -369,6 +369,22 @@ void game_load::evaluate_summary_string(std::stringstream& str, const config& cf
 	if(!cfg_summary["version"].empty()) {
 		str << "\n" << _("Version: ") << cfg_summary["version"];
 	}
+
+	const std::vector<std::string>& active_mods = utils::split(cfg_summary["active_mods"]);
+	if (!active_mods.empty()) {
```
Comment (jyrkive) [inline src/gui/dialogs/game_load.cpp]: Nit: Wesnoth code style doesn't use a space after `if`.

File: src/gui/dialogs/game_load.cpp [hunk 2]
```diff
@@ -369,6 +369,22 @@ void game_load::evaluate_summary_string(std::stringstream& str, const config& cf
...
+		str << "\n" << _("Modifications: ");
+		for (const auto& mod_id : active_mods) {
```
Comment (jyrkive) [inline src/gui/dialogs/game_load.cpp]: Nit: space after `for`.

File: src/gui/dialogs/game_load.cpp [hunk 3]
```diff
@@ -369,6 +369,22 @@ void game_load::evaluate_summary_string(std::stringstream& str, const config& cf
...
+			std::string mod_name;
+			try {
+				mod_name = cache_config_.find_child("modification", "id", mod_id)["name"].str();
```
Comment (jyrkive) [inline src/gui/dialogs/game_load.cpp]: `.str()` should be unnecessary, attribute values have implicit conversion to strings.

File: src/gui/dialogs/game_load.cpp (hunk 1 above)
Comment (jostephd) [reply src/gui/dialogs/game_load.cpp]: Fixed

File: src/gui/dialogs/game_load.cpp (hunk 2 above)
Comment (jostephd) [reply src/gui/dialogs/game_load.cpp]: Fixed

File: src/gui/dialogs/game_load.cpp (hunk 3 above)
Comment (jostephd) [reply src/gui/dialogs/game_load.cpp]: Without the `.str()` it doesn't compile.

```
src/gui/dialogs/game_load.cpp:382:77: error: ambiguous overload for ‘operator=’ (operand types are ‘std::__cxx11::string {aka std::__cxx11::basic_string<char>}’ and ‘const attribute_value {aka const config_attribute_value}’)
     mod_name = cache_config_.find_child("modification", "id", mod_id)["name"];
```

**Comment by gfgtdf** [comment]
no, invalidate_layout must be avoided at least in the current gui2 implementation, in the worst case it causes widgets to 'jump around' which makes a really crappy impression to the user, also it can be quite slow if the dialogue is complicated ( at least that's the state from when I checked this stuff last time.) , If you 'need' a relayout  then the is prevails something else wrong or you are using gui2 in a way that it was not designed to.

Ideally a (usable) relayout would only layout the content on the highest fixes-size widgets that the 'problematic' widgets is in. or rather,  we should add a 'layout blocker' widget that makes layout calls only relayout it's content. when relayout is calls inside.

**Comment by jyrkive** [comment]
A relayout is always needed when any GUI2 widget grows, that's how GUI2 is designed.

In some cases you can get away by firing a `REQUEST_PLACEMENT` event instead (but even it will fall back to full relayout if necessary).

**Pull request merged**